bridge/src/lib.rs

```diff
@@ -91,17 +92,21 @@ methods!(
         let request = unwrap_bytes(request.map_err(VM::raise_ex).unwrap());
         let metadata = to_hash_map(metadata.map_err(VM::raise_ex).unwrap());
         let timeout = timeout.map_or(None, |v| Some(v.to_u64()));
+        let (tx, rx) = mpsc::channel::<RpcResult>();
 
-        let result = Thread::call_without_gvl(move || {
+        let result = Thread::call_without_gvl(|| {
             let connection = _rtself.get_data_mut(&*CONNECTION_WRAPPER);
             let params = RpcParams {
                 rpc: rpc.clone(),
                 request: request.clone(),
                 metadata: metadata.clone(),
                 timeout_millis: timeout
             };
-            connection.call(params)
-        }, Some(|| {}));
+            connection.call(params, tx.clone());
+            rx.recv().expect("RPC result channel closed")
+        }, Some(|| {
```

Comment:
Question - how long will Ruby wait until "trying again" and/or segfaulting like you saw before?

Reply:
From what I've noticed sometimes it just hangs forever (waited for a minute and it was still hanging), but reliably segfaults on a second attempt to raise within than thread.

Reply:
Can you replicate the segfault after this change? I just want to make sure we can never segfault, it scares me.

Reply:
Nope, what was reliably seg-faulting before won't seg-fault again